models/mailchimp/mailchimp_emails_denormalized.sql

```diff
@@ -0,0 +1,53 @@
+with email_actions as (
+    select *
+    from {{env.schema}}.mailchimp_email_actions
+),
+
+sends as (
+    select campaign_id, email_id, sent_date
+    from {{env.schema}}.mailchimp_sent_to
+),
+
+bounces as
+(
+    select campaign_id, email_id, min(action_date) as bounced_date
+    from email_actions
+    where action = 'bounce'
+    group by campaign_id, email_id
+),
+
+opens as (
+    -- look at the first open date
+    select campaign_id, email_id, min(action_date) as opened_date
+    from email_actions
+    where action = 'open'
+    group by campaign_id, email_id
+),
+
+clicks as
+(
+    -- look at the first click data
+    select campaign_id, email_id, min(action_date) as clicked_date
+    from email_actions
+    where action = 'click'
+    group by campaign_id, email_id
+)
```

Comment:
I think that these CTEs (including unsubscribes) are insanely useful as models and will need to be individually accessed in subsequent analysis. I'd love to see these broken out like this:
* one model for each email action (send, open, click, bounce, unsubscribe)
* one model that unions all of those event actions together with a single field for `event_action`. we'd subsequently use this table like an events table for things like cohort and funnel analysis.
* one model that summarizes by email_id (like the below query). that table could include the summarization by email statistics (distinct opens, etc).